src/OrchardCore/OrchardCore.Localization.Core/DataAnnotations/LocalizedValidationMetadataProvider.cs

```diff
@@ -30,25 +30,36 @@ public void CreateValidationMetadata(ValidationMetadataProviderContext context)
         {
             foreach (var metadata in context.ValidationMetadata.ValidatorMetadata)
             {
-                var attribute = metadata as ValidationAttribute;
-                var displayName = context.Attributes.OfType<DisplayAttribute>().FirstOrDefault()?.Name;
-                // Use DisplayName if present
-                var argument = displayName ?? context.Key.Name;
-                var errorMessageString = attribute != null && attribute.ErrorMessage == null && attribute.ErrorMessageResourceName == null
-                    ? attribute.FormatErrorMessage(argument)
-                    : attribute.ErrorMessage;
+                if (metadata is ValidationAttribute attribute)
+                {
+                    var displayName = context.Attributes.OfType<DisplayAttribute>().FirstOrDefault()?.Name;
+                    // Use DisplayName if present
+                    var argument = displayName ?? context.Key.Name;
+                    var errorMessageString = attribute.ErrorMessage == null && attribute.ErrorMessageResourceName == null
+                        ? attribute.FormatErrorMessage(argument)
+                        : attribute.ErrorMessage;
 
-                // Localize the error message without params
-                var localizedString = _stringLocalizer[errorMessageString];
+                    // Localize the parameterized error message
+                    var localizedString = _stringLocalizer[errorMessageString];
```

Comment:
Where the params in localizer?!!

Reply:
The error message will be parameterized at this point if we set `errorMessageString = attribute.FormatErrorMessage(argument)` above.

Reply:
Right, but I remembered did that for a reason, I may do many tests before I move your changes to my PR, thanks a lot